compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolEnv.java

```diff
@@ -222,13 +222,15 @@ public static SymbolEnv createRetryEnv(BLangRetry node, SymbolEnv env) {
     }
 
     public static SymbolEnv createOnFailEnv(BLangOnFailClause node, SymbolEnv env) {
```

Comment:
Not exactly related to this PR. But if the on fail clause has a block stmt in it, do we need to be creating a separate env like this? Wouldn't it automatically be handled when analyzing the block stmt?

Reply:
Seems when the node changed, the environment creation was not reflected. Updated in b12ab10395f186458914384cb49212a460a18833

Reply:
It seems unless the OnFailClause is used to create the symbol environment, the variable definition symbol within on fail clause needs to be added manually while desugaring it to a lambda function. Hence reverted this change